include/odp/api/spec/pool.h

```diff
@@ -172,7 +172,20 @@ int odp_pool_capability(odp_pool_capability_t *capa);
 typedef struct odp_pool_param_t {
 	/** Pool type */
 	int type;
-
+	/** Flag to indicate whether user is requesting for contiguous
+	    memory for buffers or packets in pool.
+	    If TRUE, then address range for buffers or packets
+	    must be returned by API as addr_start & addr_len.
+	    If FALSE, then filling these parameters is optional.
+	 */
+	odp_bool_t contig_mem;
```

Comment:
Isn't setting `seg_len` enough? Maybe we should provide an updated definition for `seg_len`  (e.g. in case `seg_len > max_len`, always use single segment).

Reply:
Yes the existing `seg_len` parameter is set to the minimum size segment that the application requires. This can be set to be equal to `max_len` to require single-segment packets.

However @sachin-saxena mentioned at SFO17 that VPP can deal with multi-segment packets, though I'm not sure how that works.

Reply:
Got confused between single-segment packet and linear space for packets, sorry.